src/lib/linux-api/src/signal.rs

```diff
@@ -0,0 +1,548 @@
+
+impl linux_siginfo_t {
+    fn as_bound_type(&self) -> &bindings::siginfo_t {
+        static_assertions::assert_eq_align!(linux_siginfo_t, bindings::siginfo_t);
+        static_assertions::assert_eq_size!(linux_siginfo_t, bindings::siginfo_t);
+        unsafe { core::mem::transmute(self) }
+    }
+
+    fn as_bound_type_mut(&mut self) -> &mut bindings::siginfo_t {
+        static_assertions::assert_eq_align!(linux_siginfo_t, bindings::siginfo_t);
+        static_assertions::assert_eq_size!(linux_siginfo_t, bindings::siginfo_t);
+        unsafe { core::mem::transmute(self) }
+    }
+
+    pub fn signo(&self) -> &i32 {
+        unsafe {
+            &self
+                .as_bound_type()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_signo
+        }
+    }
+
+    pub fn signo_mut(&mut self) -> &mut i32 {
+        unsafe {
+            &mut self
+                .as_bound_type_mut()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_signo
+        }
+    }
+
+    pub fn errno(&self) -> &i32 {
+        unsafe {
+            &self
+                .as_bound_type()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_errno
+        }
+    }
+
+    pub fn errno_mut(&mut self) -> &mut i32 {
+        unsafe {
+            &mut self
+                .as_bound_type_mut()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_errno
+        }
+    }
+
+    pub fn code(&self) -> &i32 {
+        unsafe {
+            &self
+                .as_bound_type()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_code
+        }
+    }
+
+    pub fn code_mut(&mut self) -> &mut i32 {
+        unsafe {
+            &mut self
+                .as_bound_type_mut()
+                .__bindgen_anon_1
+                .__bindgen_anon_1
+                .si_code
+        }
+    }
+}
+
+/// Compatible with the Linux kernel's definition of sigset_t on x86_64.
+///
+/// This is analagous to, but typically smaller than, libc's sigset_t.
+#[repr(C)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Default, VirtualAddressSpaceIndependent)]
+pub struct linux_sigset_t {
+    val: u64,
+}
+static_assertions::assert_eq_align!(linux_sigset_t, bindings::sigset_t);
+static_assertions::assert_eq_size!(linux_sigset_t, bindings::sigset_t);
+
+impl linux_sigset_t {
+    pub const EMPTY: Self = Self { val: 0 };
+    pub const FULL: Self = Self { val: !0 };
+
+    pub fn has(&self, sig: LinuxSignal) -> bool {
+        (*self & linux_sigset_t::from(sig)).val != 0
+    }
+
+    pub fn lowest(&self) -> Option<LinuxSignal> {
+        if self.val == 0 {
+            return None;
+        }
+        for i in 1..=LINUX_SIGRT_MAX {
+            let s = LinuxSignal::try_from(i).unwrap();
+            if self.has(s) {
+                return Some(s);
+            }
+        }
+        unreachable!("");
```

Comment:
I think you can just do `unreachable!()`.

Reply:
done